contrib/kind.sh

```diff
@@ -146,57 +155,132 @@ print_params
-# Detect IP to use as API server
-API_IPV4=""
-if [ "$KIND_IPV4_SUPPORT" == true ]; then
-  # ip -4 addr -> Run ip command for IPv4
-  # grep -oP '(?<=inet\s)\d+(\.\d+){3}' -> Use only the lines with the
-  #   IPv4 Addresses and strip off the trailing subnet mask, /xx
-  # grep -v "127.0.0.1" -> Remove local host
-  # head -n 1 -> Of the remaining, use first entry
-  API_IPV4=$(ip -4 addr | grep -oP '(?<=inet\s)\d+(\.\d+){3}' | grep -v "127.0.0.1" | head -n 1)
-  if [ -z "$API_IPV4" ]; then
-    echo "Error detecting machine IPv4 to use as API server"
-    exit 1
+check_ipv6() {
+  # Collect additional IPv6 data on test environment
+  ERROR_FOUND=false
+  TMPVAR=`sysctl net.ipv6.conf.all.forwarding | awk '{print $3}'`
+  echo "net.ipv6.conf.all.forwarding is equal to $TMPVAR"
+  if [ "$TMPVAR" != 1 ]; then
+    if [ "$KIND_UPDATE_SYSTEM" == true ]; then
+      sudo sysctl -w net.ipv6.conf.all.forwarding=1
+    else
+      echo "RUN: 'sudo sysctl -w net.ipv6.conf.all.forwarding=1' to use IPv6."
+      ERROR_FOUND=true
+    fi
   fi
-fi
+  TMPVAR=`sysctl net.ipv6.conf.all.disable_ipv6 | awk '{print $3}'`
+  echo "net.ipv6.conf.all.disable_ipv6 is equal to $TMPVAR"
+  if [ "$TMPVAR" != 0 ]; then
+    if [ "$KIND_UPDATE_SYSTEM" == true ]; then
+      sudo sysctl -w net.ipv6.conf.all.disable_ipv6=0
+    else
+      echo "RUN: 'sudo sysctl -w net.ipv6.conf.all.disable_ipv6=0' to use IPv6."
+      ERROR_FOUND=true
+    fi
+  fi
+  if [ "$ERROR_FOUND" == true ]; then 
+    exit 1  
+  fi  
 
-API_IPV6=""
-if [ "$KIND_IPV6_SUPPORT" == true ]; then
-  # ip -6 addr -> Run ip command for IPv6
-  # grep "inet6" -> Use only the lines with the IPv6 Address
-  # sed 's@/.*@@g' -> Strip off the trailing subnet mask, /xx
-  # grep -v "^::1$" -> Remove local host
-  # sed '/^fe80:/ d' -> Remove Link-Local Addresses
-  # head -n 1 -> Of the remaining, use first entry
-  API_IPV6=$(ip -6 addr  | grep "inet6" | awk -F' ' '{print $2}' | \
-             sed 's@/.*@@g' | grep -v "^::1$" | sed '/^fe80:/ d' | head -n 1)
-  if [ -z "$API_IPV6" ]; then
-    echo "Error detecting machine IPv6 to use as API server"
+  # BEGIN - REMOVE when KIND v0.8.1 is supported
+  RESTART_DOCKER=false
+  DOCKER_DAEMON_FILE=/etc/docker/daemon.json
+  if [ -f $DOCKER_DAEMON_FILE ]; then
+    DOCKER_APPEND_STR=""
+    if ! grep -q "\"ipv6\": true" $DOCKER_DAEMON_FILE; then
+      echo "'\"ipv6\": true' NOT found!"
+      DOCKER_APPEND_STR="${DOCKER_APPEND_STR} \"ipv6\": true,"
+    fi
+    if ! grep -q "\"fixed-cidr-v6\":" $DOCKER_DAEMON_FILE ; then
+      echo "'\"fixed-cidr-v6\":' NOT found!"
+      DOCKER_APPEND_STR="${DOCKER_APPEND_STR} \"fixed-cidr-v6\": \"2001:db8:1::/64\","
+    fi
+
+    if [ "$DOCKER_APPEND_STR" != "" ]; then
+      if [ "$KIND_UPDATE_SYSTEM" == true ]; then
+        echo "Updating '$DOCKER_DAEMON_FILE' with DOCKER_APPEND_STR=$DOCKER_APPEND_STR"
+        RESTART_DOCKER=true
+        # The awk command:
+        #   Matches on '{' at the beginning of a line via: /^{/
```

Comment:
heh json in bash is fun

Reply:
consider using `jq`...

Reply:
Moved to KIND v0.8.1 on master, so this section has been removed.